### PR TITLE
DOC: Rename \doxygen/\subdoxygen aliases to \itkref/\itksubref

### PR DESCRIPTION
**Pull request by hjmjohnson**
Renames the Doxygen class cross-reference aliases `\doxygen` → `\itkref` and `\subdoxygen` → `\itksubref` toolkit-wide, keeping the old names as deprecated synonyms for out-of-tree code and external tools. Split out of #6411 (merged) so that PR stayed a focused warning-clear; now rebased onto `main`, the diff shows only the rename. **Companion: InsightSoftwareConsortium/ITKSoftwareGuide#235** defines the new LaTeX macros for the Software Guide pipeline and must merge first.

<details>
<summary>What this changes</summary>

In `Utilities/Doxygen/DoxygenConfig.cmake`:
- Adds `\itkref{1}=\ref \1` — in-tree spelling of the single-class reference.
- Adds `\itksubref{2}=\ref itk::\1::\2` — in-tree spelling of the namespaced-class reference (`\itksubref{Statistics}{Histogram}` → `itk::Statistics::Histogram`).
- Keeps `\doxygen{1}` and `\subdoxygen{2}` as deprecated synonyms (marked with a comment).

Conversions across 250 files:
- `\doxygen{...}` → `\itkref{...}` — 816 uses (221 files; 221 in `Examples/`, 20 in `Modules/`).
- `\subdoxygen{...}{...}` → `\itksubref{...}{...}` — 85 uses (31 `Examples/` files).

Mechanical, comment-only change (the aliases appear only in Doxygen doc comments). `pre-commit run --all-files` is clean.

</details>

<details>
<summary>Why / notes</summary>

Reviewer request on #6411 (albert-github): the `\doxygen` alias name is too generic. `\itkref` / `\itksubref` read as ITK cross-references and free the old names.

`\subdoxygen` was previously **never defined** in `DoxygenConfig.cmake` despite being used in-tree (Examples) and by remote modules (e.g. BioCell `\subdoxygen{bio}{Cell}`); defining it as a deprecated synonym additionally makes those existing uses resolve instead of being unknown commands. Backwards compatibility is preserved so no in-repo or out-of-tree documentation breaks.

</details>

### CODE REVIEW

**Comment by hjmjohnson** [comment]
@albert-github would you mind reviewing this, please?

File: Utilities/Doxygen/DoxygenConfig.cmake [hunk 1]
```diff
@@ -17,7 +17,11 @@ set(DOXYGEN_NUM_PROC_THREADS "0")
 set(
   DOXYGEN_ALIASES
   "starteraliasnotused=@par not used"
+  "itkref{1}=\\ref \\1"
+  "itksubref{2}=\\ref itk::\\1::\\2"
+  # Deprecated spellings of \itkref / \itksubref, kept for external tools and out-of-tree code.
   "doxygen{1}=\\ref \\1"
+  "subdoxygen{2}=\\ref itk::\\1::\\2"
```
Comment (albert-github) [inline Utilities/Doxygen/DoxygenConfig.cmake]: I'm not happy with such old `ALIASES` as they might also have influence on the ITK code as they still can be used.
I do however understand the reasoning, but don't know whether or not this will apply.
- with external tools you mean people who build their own application with some own documentation and, parts of, ITK and its documentation? I'd say a good message in the release notes would be the right thing to do and remove the definitions here.
- the out-of-tree code I don't get as here one has / uses the current ITK code so the problem should not occur.

(of course these is my personal opinion and I'm not saying the current ALIASES are wrong)

File: Examples/Segmentation/VectorConfidenceConnected.cxx [hunk 2]
```diff
@@ -37,12 +37,12 @@
...
 //  applied to images with vector pixel types. The confidence connected
 //  algorithm is implemented for vector images in the class
-//  \doxygen{VectorConfidenceConnected}. The basic difference between the
+//  \itkref{VectorConfidenceConnected}. The basic difference between the
 //  scalar and vector version is that the vector version uses the covariance
 //  matrix instead of a variance, and a vector mean instead of a scalar mean.
 //  The membership of a vector pixel value to the region is measured using the
 //  Mahalanobis distance as implemented in the class
-//  \subdoxygen{Statistics}{MahalanobisDistanceThresholdImageFunction}.
+//  \itksubref{Statistics}{MahalanobisDistanceThresholdImageFunction}.
```
Comment (albert-github) [inline Examples/Segmentation/VectorConfidenceConnected.cxx:45]:
```suggestion
//  \itksubref{Statistics,MahalanobisDistanceThresholdImageFunction}.
```

The used syntax is incorrect, also in the original, but probably not used as there are just regular comment signs (`//`) and not doxygen comment signs (`///`) present or is there some preprocessing tool running that converts the comment.

File: Examples/Segmentation/WatershedSegmentation1.cxx [hunk 3]
```diff
@@ -176,10 +176,10 @@ main(int argc, char * argv[])
   // for the purposes of this example, we will convert it to RGB pixels.  RGB
   // images have the advantage that they can be saved as a simple png file
   // and viewed using any standard image viewer software.  The
-  // \subdoxygen{Functor}{ScalarToRGBPixelFunctor} class is a special
+  // \itksubref{Functor}{ScalarToRGBPixelFunctor} class is a special
```
Comment (albert-github) [inline Examples/Segmentation/WatershedSegmentation1.cxx:179]:
```suggestion
  // \itksubref{Functor,ScalarToRGBPixelFunctor} class is a special
```

File: Examples/Statistics/BayesianPluginClassifier.cxx [hunk 4]
```diff
@@ -28,12 +28,12 @@
 // all the components of the classifier system and the data flow. This system
 // differs with the previous k-means clustering algorithms in several
 // ways. The biggest difference is that this classifier uses the
-// \subdoxygen{Statistics}{GaussianDensityFunction}s as membership functions
+// \itksubref{Statistics}{GaussianDensityFunction}s as membership functions
```
Comment (albert-github) [inline Examples/Statistics/BayesianPluginClassifier.cxx:31]: Too many to make everywhere suggestions ...

File: Examples/Statistics/ExpectationMaximizationMixtureModelEstimator.cxx [hunk 5]
```diff
@@ -66,13 +66,13 @@
...
 // Binomial, Exponential, and Normal distributions have analytical
 // solutions for updating the parameter set. The
-// \subdoxygen{Statistics}{ExpectationMaximizationMixtureModelEstimator}
+// \itksubref{Statistics}{ExpectationMaximizationMixtureModelEstimator}
 // class assumes that such type of components.
 //
-// In the following example we use the \subdoxygen{Statistics}{ListSample} as
-// the sample (test and training). The \subdoxygen{Vector} is our measurement
+// In the following example we use the \itksubref{Statistics}{ListSample} as
+// the sample (test and training). The \itksubref{Vector} is our measurement
```
Comment (albert-github) [inline Examples/Statistics/ExpectationMaximizationMixtureModelEstimator.cxx]:
```suggestion
// the sample (test and training). The \itkref{Vector} is our measurement
```

Just 1 argument present.

**Review by albert-github — COMMENTED** [review]
Looks in principle OK just some remarks as noted in the comments etc.
Seen the shear size of the PR I think a doxygen run would also be beneficial to run on the code.

File: Utilities/Doxygen/DoxygenConfig.cmake (hunk 1 above)
Comment (hjmjohnson) [reply Utilities/Doxygen/DoxygenConfig.cmake]: Remote modules also use this file. We do not have control of all remote modules, so it would break if the alias suddenly changed.  It is only for backward compatibility.

File: Examples/Segmentation/VectorConfidenceConnected.cxx (hunk 2 above)
Comment (hjmjohnson) [reply Examples/Segmentation/VectorConfidenceConnected.cxx:45]: These Software Guide blocks are consumed by the ITKSoftwareGuide LaTeX pipeline (your preprocessing suspicion was right): `SoftwareGuideConfiguration.tex.in` defines `\subdoxygen` as a two-argument `\newcommand`, so the `{A}{B}` form is the required syntax there. Doxygen itself never parses `Examples/` (its INPUT is the formatted module headers), so the comma form would only apply to uses inside `Modules/`, of which there are currently none.

File: Examples/Segmentation/WatershedSegmentation1.cxx (hunk 3 above)
Comment (hjmjohnson) [reply Examples/Segmentation/WatershedSegmentation1.cxx:179]: These Software Guide blocks are consumed by the ITKSoftwareGuide LaTeX pipeline (your preprocessing suspicion was right): `SoftwareGuideConfiguration.tex.in` defines `\subdoxygen` as a two-argument `\newcommand`, so the `{A}{B}` form is the required syntax there. Doxygen itself never parses `Examples/` (its INPUT is the formatted module headers), so the comma form would only apply to uses inside `Modules/`, of which there are currently none.

File: Examples/Statistics/BayesianPluginClassifier.cxx (hunk 4 above)
Comment (hjmjohnson) [reply Examples/Statistics/BayesianPluginClassifier.cxx:31]: These Software Guide blocks are consumed by the ITKSoftwareGuide LaTeX pipeline (your preprocessing suspicion was right): `SoftwareGuideConfiguration.tex.in` defines `\subdoxygen` as a two-argument `\newcommand`, so the `{A}{B}` form is the required syntax there. Doxygen itself never parses `Examples/` (its INPUT is the formatted module headers), so the comma form would only apply to uses inside `Modules/`, of which there are currently none.

File: Examples/Statistics/ExpectationMaximizationMixtureModelEstimator.cxx (hunk 5 above)
Comment (hjmjohnson) [reply Examples/Statistics/ExpectationMaximizationMixtureModelEstimator.cxx]: Fixed in 0c477a68d5 — applied your suggestion (`\itkref{Vector}`). This was the only single-argument `\itksubref` in the tree; as a two-argument LaTeX macro it would have swallowed the following token in the Software Guide build.

File: Utilities/Doxygen/DoxygenConfig.cmake (hunk 1 above)
Comment (hjmjohnson) [reply Utilities/Doxygen/DoxygenConfig.cmake]: @dzenanz Do you think that now is the time we should break downstream consumers of Utilities/Doxygen/DoxygenConfig.cmake and force them to update to the new aliases (migration guide?  What is your feelings on the deprecation strategy for the `doxygen` and `subdoxygen` commands?  Perhaps we should "poison pill" those aliases with something like 

"\doxygen{1}=error:  the old `\doxygen{}` macro in Utilities/Doxygen/DoxygenConfig.cmake must be updated to `\itkref{}`  \\ref \\1".

Perhaps add a Utilities/Migration/update_doxygen_for_itkv6.py script.

File: Utilities/Doxygen/DoxygenConfig.cmake [hunk 6]
```diff
@@ -17,7 +17,11 @@ set(DOXYGEN_NUM_PROC_THREADS "0")
 set(
   DOXYGEN_ALIASES
   "starteraliasnotused=@par not used"
+  "itkref{1}=\\ref \\1"
+  "itksubref{2}=\\ref itk::\\1::\\2"
```
Comment (albert-github) [inline Utilities/Doxygen/DoxygenConfig.cmake:22]: It is not necessary to have a new command here `"itksubref{2}=\\ref itk::\\1::\\2"` could also read `"itkref{2}=\\ref itk::\\1::\\2"` and doxygen will do the work.

**Review by albert-github — COMMENTED** [review]
Was still in my cache ....

**Comment by hjmjohnson** [comment]
Companion PR for the Software Guide LaTeX side: InsightSoftwareConsortium/ITKSoftwareGuide#235 defines `\itkref`/`\itksubref` (keeping the old spellings as aliases). It should merge before this PR so Guide builds keep working across the rename.

File: Examples/Segmentation/WatershedSegmentation1.cxx (hunk 3 above)
Comment (albert-github) [reply Examples/Segmentation/WatershedSegmentation1.cxx:179]: Seen this comment, my comment of redefining `\itksubref{2}` to `\itkref{2}` is not valid anymore.
Maybe one has to think about a HTML version of the ITKSoftwareGuide with the examples but this will probably be quite a bit of work.

**Comment by hjmjohnson** [comment]
Ran the verification Doxygen pass @albert-github suggested: warning output is **byte-identical** between this branch and `upstream/main` — the rename introduces zero Doxygen warning delta, as expected from alias-name-only changes with identical `\ref` expansions.

<details>
<summary>Method and timing</summary>

- Doxygen 1.13.2, warnings-only configuration: the CMake-generated `Doxyfile.Documentation` plus `GENERATE_HTML=NO`, `GENERATE_LATEX=NO`, `HAVE_DOT=NO`, `WARN_LOGFILE=…`.
- `Examples.dox` regenerated per branch via `GenerateExamplesDox.cmake`.
- Both runs emit the same 145 warning lines (pre-existing, environment-local: missing ExternalData test files and similar — none reference the aliases).
- `diff main-warnings pr-warnings` → empty.
- Wall time ≈ 63–64 s per run, 1.8 GB peak RSS — small enough that a per-PR `doxygen-warnings` GitHub Actions job is practical (~5–8 min end-to-end including checkout + configure on a standard runner); candidate follow-up.

</details>

**Review by albert-github — APPROVED** [review]

File: Utilities/Doxygen/DoxygenConfig.cmake (hunk 6 above)
Comment (hjmjohnson) [reply Utilities/Doxygen/DoxygenConfig.cmake:22]: You are right that Doxygen overloads aliases by argument count, and a single `\itkref{1}`/`\itkref{2}` pair would work on the Doxygen side. We are keeping the two names because the same commands must also exist as **LaTeX macros** in the ITK Software Guide (InsightSoftwareConsortium/ITKSoftwareGuide#235), which consumes the `Examples/` blocks: plain LaTeX cannot overload a `\newcommand` by arity, and the single-name alternative there is an `\@ifnextchar\bgroup` lookahead macro whose brace-group peeking fails silently around whitespace/line breaks. Two plainly-defined names keep arity explicit in both pipelines and make misuse fail loudly (as the `\itksubref{Vector}` case you caught did). Adding an inline comment in `DoxygenConfig.cmake` recording this rationale with the next push.

File: Utilities/Doxygen/DoxygenConfig.cmake (hunk 1 above)
Comment (hjmjohnson) [reply Utilities/Doxygen/DoxygenConfig.cmake]: Implemented in 1acdf7e3dd: the deprecated aliases now render an inline "error: deprecated ... alias" marker ahead of the still-working reference, and Utilities/ITKMigrationPreparation/update_doxygen_for_itkv6.py performs the rewrite (validated idempotent; a dry-run over the cloned remote modules found 83 sites in 11 files that will need it).

**Review by dzenanz — APPROVED** [review]

**Comment by hjmjohnson** [comment]
@greptileai review
